crates/sui/src/keytool.rs

```diff
@@ -339,36 +339,37 @@ impl KeyToolCommand {
             }
 
             KeyToolCommand::Import {
-                mnemonic_phrase,
+                input_string,
                 key_scheme,
                 derivation_path,
             } => {
-                let address =
-                    keystore.import_from_mnemonic(&mnemonic_phrase, key_scheme, derivation_path)?;
-                info!("Key imported for address [{address}]");
+                // check if it is a private key
+                if input_string.starts_with("0x") {
+                    if input_string.len() != 66 {
```

Comment:
you dont need to check the string, you can just use `let bytes = Hex::decode(input_string)?` which should takes care of the parsing from hex -> bytes. 

then instead of using the magic number 66 you can just check `let kp = Ed25519KeyPair::from_bytes(bytes)?` which does the length validity check implicitly and gives you a valid ed25519private

you can then just get a wellformed skp via `let skp = SuiKeyPair::Ed25519(kp)` then `keystore.add_key(skp)`

Reply:
Awesome, thank you! So the wallet only supports `ed25519` at the moment. If the user provides a different scheme when importing a private key, not sure how to handle that? Any suggestions here?

Reply:
then you need to specify what scheme you intend to usse when calling this cli. or alternatively the hex value needs to encode the flag (which the private key value in chrome extension does not include)